source/slang/slang-check-impl.h

```diff
@@ -943,6 +955,13 @@ namespace Slang
         ASTBuilder* m_astBuilder = nullptr;
 
         Scope* m_outerScope = nullptr;
+
+        // By default, we will support short-circuit evaluation for the logic expression.
+        // However, there are few exceptions where we will disable it:
+        // 1. the logic expression is inside the generic parameter list.
+        // 2. the logic expression is in the init expression of a static const variable.
+        // 3. the logic expression is in the index of array.
```

Comment:
the expression is a size of an array type, or we can just remove the comment because I assume no addiitional work is required to ensure this is the case?

Reply:
update the comment.